stablehlo/dialect/TypeInference.cpp

```diff
@@ -2391,10 +2391,24 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
 }
 
 LogicalResult inferSortOp(
-    Optional<Location>, ValueRange inputs,
+    Optional<Location> location, ValueRange inputs,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
-  for (auto resultType : inputs.getTypes())
-    inferredReturnShapes.emplace_back(resultType.cast<ShapedType>());
```

Comment:
Given the offline discussion, let's update the PR to just emplace_back the components of `resultType`. Unfortunately `emplace_back(resultType)` won't work because it doesn't account for encoding, so we'll need to explicitly call out the three components.